mitodl/ocw-studio
# GDrive File Size Sync

# What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/1682

# Description (What does it do?)
It updates the DriveFile model to add `size` field and also updates WebsiteContent's metadata to reflect the size in `file_size`.

# How can this be tested?
1. Navigate to your local clone of ocw-hugo-projects. Checkout to main/master branch, and pull from it.
2. Navigate to your local setup of ocw-studio and checkout this branch.
3. Start OCW Studio
4. 
```
docker-compose exec web ./manage.py override_site_config -s ocw-course-v2 -c path/to/ocw-hugo-projects/ocw-course-v2/ocw-studio.yaml
docker-compose exec web ./manage.py override_site_config -s ocw-www -c path/to/ocw-hugo-projects/ocw-www/ocw-studio.yaml
```
5. Run  `docker-compose exec web ./manage.py migrate`
6. Run  `docker-compose exec web ./manage.py generate_example_configs`
7. Open a course or www site.
   **Expected Behavior**: The UI is working. Specifically the resource-related UI elements.
8. Add or update a resource.
9. Run
    ```
    docker-compose exec web ./manage.py backpopulate_pipelines --filter <id>
    ```
10. Publish the test site.
11. **Expected Behavior**: The backend (GitHub) content file for your resource has a `file_size` field with populated file size.
12. **Expected Behavior**: Publishing succeeds.

# Run tests:
The changes in this PR can have an impact on a lot of tests, like site-config tests (because of config changes), video tests (because of added file_size), DriveFile tests, WebsiteContent tests, etc. 
So for sake of thoroughness, we will run pytest for all tests:
`docker-compose run --rm web pytest
`
